.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: Build
+
+on:
+  pull_request:
+    branches: [ master, release/** ]
+  push:
+    branches: [ master, release/** ]
+    tags: [ v* ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: fcfs
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: swirlds
+          PGDATA: /var/lib/postgresql/data/pgdata
+          POSTGRES_PORT: 5432
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: ./mvnw install
```

Comment:
It would be nice to integrate sonar here. Assuming the secrets are already in place (check with Michael), you should just be able to append the `sonar:sonar` plugin. Check the mirror node for an [example](https://github.com/hashgraph/hedera-mirror-node/blob/f3d86d56042cf975b16a8d343d5998fc0971a7cf/.github/workflows/security.yml#L28).

It should probably be a separate step that's conditional on a non-empty secret so it doesn't fail for outside contributors, though.